### PR TITLE
Enable debugging tracing of SQL queries

### LINKED ISSUE #2389: Enable debugging tracing of SQL queries
### What should be done?

After #1717.

In debug builds only (`if debugbuild.Enabled`), enable pgx/v5 tracing integration to send traces to our development Jaeger instance. Do not add flags for configuring Jaeger address, etc – that's all only for us during development.

### Where?

`pgdb`

### Definition of Done

- manual testing (just run integration tests, see traces);
- spot refactorings done.

### PR DESCRIPTION
**Pull request by craigpastro**
# Description

Closes #2389.

I had a quick go at trying to resolve #2389. Is this the approach you would like to take? I can see the spans in Jaeger, but, afaict, they are not part of anything else.

Screenshot 2023-04-18 at 10 10 22

## Readiness checklist

* [ ] I added/updated unit tests.
* [ ] I added/updated integration/compatibility tests.
* [ ] I added/updated comments and checked rendering.
* [ ] I made spot refactorings.
* [ ] I updated user documentation.
* [ ] I ran `task all`, and it passed.
* [ ] I ensured that PR title is good enough for the changelog.
* [ ] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
* [ ] I marked all done items in this checklist.

### CODE REVIEW

File: internal/handlers/pg/pgdb/trace.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+// TraceQueryStart adds a span to Query, QueryRow, and Exec calls.
+func (t *debugTracer) TraceQueryStart(ctx context.Context, _ *pgx.Conn, _ pgx.TraceQueryStartData) context.Context {
+	ctx, _ = tracer.Start(ctx, "TraceQueryStart")
+	return ctx
+}
+
+// TraceQueryEnd ends the span started by TraceQueryStart.
+func (t *debugTracer) TraceQueryEnd(ctx context.Context, _ *pgx.Conn, _ pgx.TraceQueryEndData) {
+	span := trace.SpanFromContext(ctx)
```
Comment (rumyantseva) [inline internal/handlers/pg/pgdb/trace.go:39]: Same here: let's use some data from `TraceQueryEndData` as span attributes to have more info available on the Jaeger UI.

**Review by rumyantseva — CHANGES_REQUESTED** [review]
Thank you very much for sending the PR!

Yes, this approach is something we want to start from. Of course, we want to add an ability to trace requests, so those `pgdb` spans will become parts of some other spans. But it's for a separate issue. For now, we just want to start seeing at least some data :)

I added a couple of comments about `TraceQueryStart` and `TraceQueryEnd` implementation - let's add some data as span attributes, so we can see some actual info about the queries in Jaeger.

File: internal/handlers/pg/pgdb/trace.go (hunk 1 above)
Comment (craigpastro) [reply internal/handlers/pg/pgdb/trace.go:44]: Done. Thank you!

File: internal/handlers/pg/pgdb/trace.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+var tracer = otel.Tracer("internal/handlers/pg/pgdb")
+
+// debugTracer implements pgx.QueryTracer. It is used to add traces to Query,
+// QueryRow, and Exec calls in debug builds.
+type debugTracer struct{}
+
+// TraceQueryStart adds a span to Query, QueryRow, and Exec calls.
+func (t *debugTracer) TraceQueryStart(ctx context.Context, _ *pgx.Conn, _ pgx.TraceQueryStartData) context.Context {
+	ctx, _ = tracer.Start(ctx, "TraceQueryStart")
```
[the remark this answers is not in the document]
Comment (rumyantseva) [reply internal/handlers/pg/pgdb/trace.go]: I like your approach with `%#v`, yes! Thank you!

[the remark this answers is not in the document]
Comment (craigpastro) [reply internal/handlers/pg/pgdb/trace.go]: Updated. Thanks!

**Review by w84thesun — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**